src/c_api/init.cpp

```diff
@@ -0,0 +1,47 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2023, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#include <dlaf_c/init.h>
+
+#include <pika/execution.hpp>
+#include <pika/init.hpp>
+#include <pika/program_options.hpp>
+#include <pika/runtime.hpp>
+#include <dlaf/init.h>
+
+static bool dlaf_initialized = false;
+
+void dlaf_initialize(int argc, const char** argv) {
```

Comment:
```suggestion
extern "C" {
void dlaf_initialize(int argc, const char** argv) {
```

or my preference would be to only have the `extern "C"` on the declaration. Whichever we do, we should be consistent. Currently some of the C API _definitions_ use `extern "C"` (`dlaf_pdpotrf` etc.) and some don't (`dlaf_initialize` etc.).

Semi-related, it may be worth defining a macro:
```
#ifdef __cplusplus
#define DLAF_EXTERN_C
#else
#define DLAF_EXTERN_C extern "C"
#endif
```
and use that on the declarations instead.

Reply:
The macro is a good suggestion, thanks.